apisix/cli/ops.lua

```diff
@@ -542,6 +543,29 @@ Please modify "admin_key" in conf/config.yaml .
         end
     end
 
+    -- check control port conflict when batch-requests is enabled
+    if enabled_plugins["batch-requests"] and yaml_conf.apisix.enable_control then
+        local split = pl_stringx.split
+        local ip_port = split(sys_conf.control_server_addr, ":")
+        local port = tonumber(ip_port[#ip_port])
+        if port == nil then
+            util.die("port not number, should not happen\n")
+        end
+
+        if type(yaml_conf.apisix.node_listen) == "number" then
```

Comment:
Maybe we can move the control API stuff above: https://github.com/apache/apisix/blob/d0bc7238c912259e325f45ad6caea039d24e4292/apisix/cli/ops.lua#L410

And use a port denylist to check it when iterating the node_listen? So that we don't need to iterate it twice.

Reply:
I think so. I've commit the code.